### PR TITLE
Updating build.md with Maya 2022

### PR DESCRIPTION
**Pull request by seando-adsk**
Updating build.md with Maya 2022 and dropping Maya 2018.

### CODE REVIEW

File: doc/build.md [hunk 1]
```diff
@@ -10,18 +10,18 @@ Before building the project, consult the following table to ensure you use the r
...
 |    Command processor  | Visual Studio X64 2015 or 2017 command prompt  |                     bash                |             bash            |
-| Supported Maya Version|      2018, 2019, 2020           |                      2018, 2019, 2020                  |        2018, 2019, 2020     |
+| Supported Maya Version|     2019, 2020, 2022      |                      2019, 2020, 2022                        |      2019, 2020, 2022       |
 
 |        Optional       | ![](images/windows.png)   |                            ![](images/mac.png)               |   ![](images/linux.png)     |
 |:---------------------:|:-------------------------:|:------------------------------------------------------------:|:---------------------------:|
-|          Qt           | Maya 2018/2019 = 5.6.1<br>Maya 2020 = 5.12.5 | Maya 2018/2019 = 5.6.1<br>Maya 2020 = 5.12.5 | Maya 2018/2019 = 5.6.1<br>Maya 2020 = 5.12.5 |
+|          Qt           | Maya 2019 = 5.6.1<br>Maya 2020 = 5.12.5<br>Maya 2022 = 5.15.2 | Maya 2019 = 5.6.1<br>Maya 2020 = 5.12.5<br>Maya 2022 = 5.15.2 | Maya 2019 = 5.6.1<br>Maya 2020 = 5.12.5<br>Maya 2022 = 5.15.2 |
```
Comment (seando-adsk) [inline doc/build.md:24]: Added Maya 2022 with Qt version.

File: doc/build.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ For additional information on building Pixar USD, see the ***Additional Build In
 
 #### 3. Universal Front End (UFE)
 
-The Universal Front End (UFE) is a DCC-agnostic component that allows Maya to browse and edit data in multiple data models. This allows Maya to edit pipeline data such as USD. UFE comes installed as a built-in component with Maya 2019 and later. UFE is developed as a separate binary component, and therefore versioned separately from Maya. Current shipped version of UFE is **1.0.0**
+The Universal Front End (UFE) is a DCC-agnostic component that allows Maya to browse and edit data in multiple data models. This allows Maya to edit pipeline data such as USD. UFE comes installed as a built-in component with Maya 2019 and later. UFE is developed as a separate binary component, and therefore versioned separately from Maya. Maya 2019 and Maya 2020 both shipped with UFE **v1.0.0**. Maya 2022 shipped with UFE **v2.0.0**.
```
Comment (seando-adsk) [inline doc/build.md:42]: Updated to mention UFE v2 for Maya 2022.

File: doc/build.md [hunk 3]
```diff
@@ -187,22 +187,11 @@ Test project /Users/sabrih/Desktop/workspace/build/Debug/plugin/al
...
-##### Boost:
-
-Currently the Animal Logic plugin has a dependency on some of the boost components. When building Pixar USD, one needs to pass the following key,value paired arguments for boost to include those components: 
-e.g 
-
-```
-python build_usd.py ~/Desktop/BUILD --build-args boost,"--with-date_time" 
-```
-
-***NOTE:*** ```--build-args``` needs to be passed at the very end of command, after build/install location.
```
Comment (seando-adsk) [inline doc/build.md:199]: This is no longer required - right?

File: doc/build.md [hunk 4]
```diff
@@ -10,18 +10,18 @@ Before building the project, consult the following table to ensure you use the r
 
 |        Required       | ![](images/windows.png)   |                            ![](images/mac.png)               |   ![](images/linux.png)     |
 |:---------------------:|:-------------------------:|:------------------------------------------------------------:|:---------------------------:|
-|    Operating System   |         Windows 10        | Catalina (10.15), Mojave (10.14), High Sierra (10.13)        |       CentOS 7              |
-|   Compiler Requirement| Maya 2018/2019 (VS 2015 update 3)<br>Maya 2020 (VS 2017) | Maya 2018/2019 (Xcode 7.3.1)<br>Maya 2020 (Xcode version 10.2.1) | Maya 2018 (gcc 4.8.2)<br>Maya 2019/2020 (gcc 6.3.1) |
-| Minimum Cmake Version |           3.13            |                             3.13                             |         3.13                |
+|    Operating System   |         Windows 10        | High Sierra (10.13)<br>Mojave (10.14)<br>Catalina (10.15)    |       CentOS 7              |
```
Comment (seando-adsk) [inline doc/build.md:13]: Here I just reordered the versions to match how the rest of the table lists things.

File: doc/build.md [hunk 5]
```diff
@@ -10,18 +10,18 @@ Before building the project, consult the following table to ensure you use the r
...
+|   Compiler Requirement| Maya 2019 (VS 2017)<br>Maya 2020 (VS 2017)<br>Maya 2022 (VS 2017) | Maya 2019 (Xcode 7.3.1)<br>Maya 2020 (Xcode 10.2.1)<br>Maya 2022 (Xcode 10.2.1) | Maya 2019 (gcc 6.3.1)<br>Maya 2020 (gcc 6.3.1)<br>Maya 2022 (gcc 6.3.1/9.3.1) |
```
Comment (seando-adsk) [inline doc/build.md:14]: Removed Maya 2018 (here and in all locations in file) and added Maya 2022. As per Krystian, changed the VS version to 2017 (from 2015) for Maya 2019 as we no longer support VS 2015. For Linux I mentioned both gcc 6 & 9.

File: doc/build.md [hunk 6]
```diff
@@ -10,18 +10,18 @@ Before building the project, consult the following table to ensure you use the r
...
+| CMake Version (min/max) |        3.13 - 3.17      |                              3.13 - 3.17                     |           3.13 - 3.17       |
```
Comment (seando-adsk) [inline doc/build.md:15]: Added min/max cmake versions to match our top-level CMakeLists.txt

**Review by HamedSabri-adsk — CHANGES_REQUESTED** [review]
- Under "Windows 10 / Command processor", still says "Visual Studio X64 2015". This should be removed since we no longer support VS 2015

- "macOS Big Sur" is missing from the list

- "Centos 8" is missing from the list

**Review by HamedSabri-adsk — APPROVED** [review]

**Pull request merged**